### DISCUSSION TITLE
Error when building latest release (3.0.0) on linux

### DISCUSSION THREAD

**Discussion by simon-r-white** [SVG-Edit/svgedit#527] [Q&A]
**Describe the bug**
Downloaded latest release [3.0.0](https://github.com/SVG-Edit/svgedit/releases/tag/v3.0.0) attempted to build, make command failed with the following error:

````
user@machine:~/svgedit-3.0.0$ make
java -jar build/tools/closure-compiler.jar \
	--compilation_level SIMPLE_OPTIMIZATIONS \
	--js editor/namespaces.js --js editor/jQuery.attr.js --js editor/contextmenu/jQuery.contextMenu.js --js editor/svgpathseg.js --js editor/browser.js --js editor/svgtransformlist.js --js editor/math.js --js editor/units.js --js editor/utilities.js --js editor/sanitize.js --js editor/history.js --js editor/historyrecording.js --js editor/coords.js --js editor/recalculate.js --js editor/select.js --js editor/draw.js --js editor/layer.js --js editor/path.js --js editor/svgcanvas.js --js editor/svg-editor.js --js editor/locale/locale.js --js editor/contextmenu.js \
	--js_output_file editor/svgedit.compiled.js
editor/namespaces.js:13: ERROR - Parse error. identifier is a reserved word
export const NS = {
       ^

editor/namespaces.js:14: ERROR - Parse error. syntax error
  HTML: 'http://www.w3.org/1999/xhtml',
        ^

editor/namespaces.js:15: ERROR - Parse error. syntax error
  MATH: 'http://www.w3.org/1998/Math/MathML',
        ^

editor/namespaces.js:16: ERROR - Parse error. syntax error
  SE: 'http://svg-edit.googlecode.com',
      ^

editor/namespaces.js:17: ERROR - Parse error. syntax error
  SVG: 'http://www.w3.org/2000/svg',
       ^

editor/namespaces.js:18: ERROR - Parse error. syntax error
  XLINK: 'http://www.w3.org/1999/xlink',
         ^

editor/namespaces.js:19: ERROR - Parse error. syntax error
  XML: 'http://www.w3.org/XML/1998/namespace',
       ^

editor/namespaces.js:20: ERROR - Parse error. syntax error
  XMLNS: 'http://www.w3.org/2000/xmlns/' // see http://www.w3.org/TR/REC-xml-names/#xmlReserved
         ^

editor/namespaces.js:27: ERROR - Parse error. identifier is a reserved word
export const getReverseNS = function () {
       ^

editor/namespaces.js:29: ERROR - Parse error. syntax error
  Object.entries(NS).forEach(([name, URI]) => {
                                              ^

editor/namespaces.js:31: ERROR - Parse error. syntax error
  });
   ^

editor/namespaces.js:32: ERROR - Parse error. invalid return
  return reverseNS;
         ^

editor/namespaces.js:33: ERROR - Parse error. syntax error
};
 ^

13 error(s), 0 warning(s)
Makefile:62: recipe for target 'editor/svgedit.compiled.js' failed
make: *** [editor/svgedit.compiled.js] Error 13

````

**Desktop (please complete the following information):**
 - OS: Linux (Ubuntu 18.04.1)

**Additional context**

Java version on machine
````
user@machine:~/$ java --version
openjdk 10.0.2 2018-07-17
OpenJDK Runtime Environment (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.2)
OpenJDK 64-Bit Server VM (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.2, mixed mode)
````

**Comment by brettz9** [comment]
The Makefile script has not been maintained, but it is still in the repo in case we end up tending again to some of its features such as using Closure (we are instead using Rollup minification now, for example). Please use npm or just use as is out of the box; there's no need for making a build in this way.

There are, as per the README, however, some cases by which one may wish to work on the source module files and rebuild files with npm scripts (such as to modify the ES6 Modules source and build non-modular versions from these source files which will work in older browsers). But one can also avoid these build steps and just directly modify the non-modular version (or the modular version, bearing in mind that the latter only works in modern browsers).

Closing, though feel free to comment further if needed.

Btw, I'm hoping to release a migration guide shortly that indicates breaking changes and what not from the past version and summarizes the changes (the CHANGES file is rather long); but the README and docs should still cover things. Also, I also just released a critical fix in version 3.0.1.

**Comment by brettz9** [comment]
FYI, I've just released 3.1.0 which among a few other tweaks, removes the Makefile. This release is not as critical as 3.0.1, however (which fixed a recent regression).

You may also be interested to know that we now have a version 3.0.0 upgrade guide and summary of changes at https://github.com/SVG-Edit/svgedit/blob/master/docs/versions/3.0.0.md#version-300 (in addition to the detailed CHANGES file listings). Feel free to ask if anything is unclear.

**Comment by simon-r-white** [comment]
Thanks... I hadn't seen the 3.0.1 release - sorry for double reporting a known regression.

Just looked at the 3.1.0 release and it is working for me.

Thanks.

**Comment by brettz9** [comment]
No worries--and you helped identify that it was not only polygons affected for a PR that would be good to reapply once we may figure out how not to cause problems like this one you reported...

Glad to hear it's working for you!
